logdevice/ops/ldquery/tables/Directories.cpp

```diff
@@ -0,0 +1,139 @@
+       "sequencer allows to be outstanding."},
+      {"backlog_duration_sec",
+       DataType::INTEGER,
+       "Time-based retention of records of logs in that directory.  If null or "
+       "zero, this directory does not use time-based retention."},
+      {"storage_set_size",
+       DataType::INTEGER,
+       "Size of the storage set for logs in that directory.  The storage set "
+       "is the set of shards that may hold data for a log."},
+      {"delivery_latency",
+       DataType::INTEGER,
+       "For logs in that directory, maximum amount of time that we can delay "
+       "delivery of newly written records.  This option increases delivery "
+       "latency but improves server and client performance."},
+      {"scd_enabled",
+       DataType::INTEGER,
+       "Indicates whether the Single Copy Delivery optimization is enabled for "
+       "this directory.  This efficiency optimization allows only one copy of "
+       "each record to be served to readers."},
+      {"custom_fields",
+       DataType::TEXT,
+       "Custom text field provided by the user."},
+  };
+}
+
+std::shared_ptr<TableData> Directories::getData(QueryContext& ctx) {
+  auto result = std::make_shared<TableData>();
+  auto full_client = ld_ctx_->getFullClient();
+
+  ld_check(full_client);
+  ClientImpl* client_impl = static_cast<ClientImpl*>(full_client.get());
+
+  auto add_row =
+      [&](const std::unique_ptr<client::Directory>& dir, const std::string dir_name, const std::string path) {
+    auto dir_attrs = dir->attrs();
+
+    ColumnValue custom;
+    if (dir_attrs.extras().hasValue() && (*dir_attrs.extras()).size() > 0) {
+      folly::dynamic extra_attrs = folly::dynamic::object;
+      for (const auto& it : dir_attrs.extras().value()) {
+        extra_attrs[it.first] = it.second;
+      }
+      custom = folly::toJson(extra_attrs);
+    }
+
+    ReplicationProperty rprop =
+        ReplicationProperty::fromLogAttributes(dir_attrs);
+    // This should remain the first ColumnValue as expected by the code below.
+    result->cols["path"].push_back(path);
+    result->cols["name"].push_back(dir_name);
+
+    result->cols["replication_property"].push_back(rprop.toString());
+    result->cols["synced_copies"].push_back(s(*dir_attrs.syncedCopies()));
+    result->cols["max_writes_in_flight"].push_back(
+        s(*dir_attrs.maxWritesInFlight()));
+    result->cols["backlog_duration_sec"].push_back(
+        s(*dir_attrs.backlogDuration()));
+    result->cols["storage_set_size"].push_back(s(*dir_attrs.nodeSetSize()));
+    result->cols["delivery_latency"].push_back(s(*dir_attrs.deliveryLatency()));
+    result->cols["scd_enabled"].push_back(s(*dir_attrs.scdEnabled()));
+    result->cols["custom_fields"].push_back(custom);
+  };
+
+  // typedef std::function<void(Status status, std::unique_ptr<client::Directory>)>
+
+  // If the query contains a constraint on the group name, we can efficiently
+  // find the directory using getLogRangeByName().
```

Comment:
`getLogRangeByName` here should be `getDirectorySync`.

Reply:
right